### PR TITLE
implement mailchimp shayan

### LINKED ISSUE #120: implement mailchimp
## Story

- **As a** .............

- **I Want** .............

- **So That** .............

## Acceptance Criteria

1. Given .............
   - When .............
   - Then .............
2. Given .............
   - When .............
   - Then .............

## Implementation Notes

1.
2.

## Checklist:

- [ ] I have defined the user story
- [ ] I have defined some acceptance criteria
- [ ] I have included any implementation notes
- [ ] I have included all relevant links

### PR DESCRIPTION
**Pull request by ShayanMahnam**
## Description

it works fine and smooth
if you like feel free to change the messages ( specifically colors )

## Related to

Fixes #120 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [x] I have commented my code
- [x] ~I have updated any documentation~

### CODE REVIEW

File: client/src/components/Footer/Footer.jsx [hunk 1]
```diff
@@ -62,12 +74,39 @@ const Footer = () => {
...
+									className="footer-subscribe-form-input"
+									placeholder="Your email"
+									value={email} // Set the value of the email input
+									onChange={(e) => setEmail(e.target.value)} // Handle input changes
+								/>
+								<button type="submit" className="footer-subscribe-form-button">
+									Subscribe
+								</button>
+								{status === "sending" && (
+									<div style={{ color: "blue" }}>sending...</div>
```
Comment (bazmurphy) [inline client/src/components/Footer/Footer.jsx]:
```suggestion
									<div className="footer-subscribe-form-status-loading">Subscribing...</div>
```

File: client/src/components/Footer/Footer.jsx [hunk 2]
```diff
@@ -62,12 +74,39 @@ const Footer = () => {
...
+								)}
+								{status === "error" && (
+									<div
+										style={{ color: "red" }}
```
Comment (bazmurphy) [inline client/src/components/Footer/Footer.jsx]:
```suggestion
										className="footer-subscribe-form-status-error"
```

File: client/src/components/Footer/Footer.jsx [hunk 3]
```diff
@@ -62,12 +74,39 @@ const Footer = () => {
...
+										dangerouslySetInnerHTML={{ __html: message }}
+									/>
+								)}
+								{status === "success" && (
+									<div style={{ color: "green" }}>Subscribed!</div>
```
Comment (bazmurphy) [inline client/src/components/Footer/Footer.jsx]:
```suggestion
									<div className="footer-subscribe-form-status-success">Subscribed!</div>
```

File: client/src/components/Footer/Footer.jsx [hunk 4]
```diff
@@ -1,5 +1,10 @@
+import { useState } from "react";
 import "./Footer.css";
 import { NavLink } from "react-router-dom";
+import MailchimpSubscribe from "react-mailchimp-subscribe";
+
+const url =
+	"https://gmail.us21.list-manage.com/subscribe/post?u=062ecc3ec4949eb98966db14b&amp;id=0664e89465&amp;f_id=00da28e7f0";
```
Comment (bazmurphy) [inline client/src/components/Footer/Footer.jsx]: Even though this is now on GitHub 

Store this in the client `.env`

as `VITE_MAILCHIMP_URL=https://gmail.us21.list-manage.com/subscribe/post?u=062ecc3ec4949eb98966db14b&amp;id=0664e89465&amp;f_id=00da28e7f0`

then use `const url = ${import.meta.env.VITE_MAILCHIMP_URL}`

and then add this to Netlify Environment Variables

(So we can transfer to them after)

File: client/src/components/Footer/Footer.jsx [hunk 5]
```diff
@@ -62,12 +74,39 @@ const Footer = () => {
...
+								<button type="submit" className="footer-subscribe-form-button">
+									Subscribe
+								</button>
+								{status === "sending" && (
+									<div style={{ color: "blue" }}>sending...</div>
+								)}
+								{status === "error" && (
+									<div
+										style={{ color: "red" }}
+										dangerouslySetInnerHTML={{ __html: message }}
```
Comment (bazmurphy) [inline client/src/components/Footer/Footer.jsx]: what value does this return? 
We might be able to parse it without dangerouslySetInnerHTML
Can you show me the shape of what it is trying to parse?

Comment (ShayanMahnam) [reply client/src/components/Footer/Footer.jsx]: write test@gmail.com you will see it

**Review by bazmurphy — APPROVED** [review]
Fixed together in Pair Programming 👍

**Pull request merged**